identity-apps-core/apps/authentication-portal/src/main/webapp/includes/language-switcher.jsp

```diff
@@ -43,7 +44,7 @@
         const localeFromCookie = getCookie("ui_lang");
         var localeFromUrlParams = null;
         if (urlParams.has('ui_locales')) {
-            localeFromUrlParams = encodeURIComponent(urlParams.get('ui_locales'));
+            localeFromUrlParams = "<%= Encode.forHtmlAttribute(request.getParameter("ui_locales")) %>";
```

Comment:
The localeFromUrlParams parameter is entirely used in JS code. Therefore the correct method to be used should be `encode.forJavaScript`

Reply:
@pavinduLakshan The computedLocale is the value we render to the html input, but the computeLocale is a javascript level code which we can't use in jsp level, so we can't encode the computedLocale. Here we encode the ui_locales value from request parameters. That's why I have encoded the localeFromUrlParams value.